File: ruby/triangle/triangle.rb
class Triangle
  def initialize(sides)
    @a, @b, @c = sides
  end

  def equilateral?
    !illogical? && a == b && b == c
  end

  def isosceles?
    !illogical? && (a == b || b == c || a == c)
  end

  def scalene?
    !illogical? && a != b && b != c && a != c
  end

  private

  attr_reader :a, :b, :c

  def illogical?
    a >= (b + c) || b >= (a + c) || c >= (a + b)
  end
end

File: clojure/triangle/src/triangle.clj
(ns triangle)

(defn illogical? [a b c]
  (or (>= a (+ b c))
      (>= b (+ a c))
      (>= c (+ a b))))

(def is-valid? (complement illogical?))

(defn equilateral? [& sides]
  (and (apply is-valid? sides)
       (= 1 (count (distinct sides)))))

(defn isosceles? [& sides]
  (and (apply is-valid? sides)
       (>= 2 (count (distinct sides)))))

(defn scalene? [& sides]
  (and (apply is-valid? sides)
       (= 3 (count (distinct sides)))))
